src/Vehicle/Vehicle.cc

```diff
@@ -4002,6 +4010,53 @@ void Vehicle::gimbalControlValue(double pitch, double yaw)
         MAV_MOUNT_MODE_MAVLINK_TARGETING);   // MAVLink Roll,Pitch,Yaw
 }
 
+void Vehicle::gimbalPitchStep(int direction)
```

Comment:
I got confused by the naming here.  It is not properly the direction. I can pass to it an integer such as 5 and as a result I get a 5 degree step.
Wouldn't be better to declare it as a bool? something like rotateClockwise?
An other option can be to extend the args like (int stepSize,bool clockwise)
This consideration apply to various function in this class.

Reply:
This is work in progress. We need new MAVLink messages to handle this. 
Here is the work on this: https://github.com/mavlink/mavlink/pull/1174

With that said, the idea when using "steps" is indicating direction (negative/positive) but also rate (1 to 100). If you want to slowly move one step in the positive direction you would send a "1". If you want to move as fast as it gets to the negative direction, you would use -100.

You have 3 scenarios:
* Single button press: Move one "step" command in one direction or another
* Button press and hold: Stream "step" messages
* Analog dial/knob: Stream current dial position messages

"Steps" are for now arbitrary. Yes, the logical mapping for a "step" for gimbal is "angle" but is it 1°? 0.1°? 10°? Hence the "step".

Reply:
>With that said, the idea when using "steps" is indicating direction (negative/positive) but also rate (1 to 100). If you want to slowly move one step in the positive direction you would send a "1". If you want to move as fast as it gets to the negative direction, you would use -100.
Yes the logic make sense. On this point I was just concerning about the naming. "Direction" seems to me a variable that can only assume +1 or -1 as values.